### ISSUE TITLE
[Bug] Form values not initialized correctly since 0.18.2/0.18.3

### ISSUE THREAD

**Issue by TheThirdRace** [bendera/vscode-commit-message-editor#40]
## Report

After a small 2 weeks break from my personal project, I returned to it and have problem with `Commit Message Editor` extention in VSCode.

The problem is **it doesn't produce the commit message correctly** if I don't change the `form` default selected values.

I use a custom configuration for the extension, details below...

Other info:
OS: Windows 10 21H1 (latest updates)
VSCode: 1.59.0 (latest version)
Commit Message Editor: 0.18.3

### How to reproduce

1. Click on the small pen in source control to display the `form` view
2. Don't touch anything beside typing description, ex: "Add blog section"
3. Click on `save and close` button
4. Message is `: Add blog section`

**Everything works as expected if I do these steps:**

1. Open each `<select>` in the `form`
2. Change their default selected value to something else
3. Change their selected value back to the default value

This is a completly new behavior. I don't know if it's since 0.18.2 or 0.18.3 as I didn't work on my project while the extension was updated twice in a short amount of time.

## Expected behavior

Same as before 0.18.2/0.18.3:

1. Click on the small pen in source control to display the `form` view
2. Don't touch anything beside typing description, ex: "Add blog section"
3. Click on `save and close` button
4. Message is `feat(content): :sparkles: Add blog section`

The previous behavior was initializing the variables in the commit message template to the first value of the form's `<select>` elements.

The workflow before the bug was very efficient. As long as you put the most used values as first value in `<select>` elements, you mostly only have to type a description.

This bug makes the current workflow very inefficient and annoying... Your extension is awesome and I have the utmost respect for authors like you, I'm just trying to convey this has real impact on users. Keep up the good work!

## VSCode `settings.json`

Here is my custom configuration:

```JSON
"commit-message-editor.dynamicTemplate": [
	"{type}{scope}: {gitmoji} {description}",
	"",
	"{body}",
	"",
	"{breaking_change}{footer}"
],
"commit-message-editor.view.defaultView": "form",
"commit-message-editor.view.saveAndClose": true,
"commit-message-editor.view.visibleViews": "form",
"commit-message-editor.tokens": [
	{
		"label": "Type",
		"name": "type",
		"description": "Type of changes",
		"type": "enum",
		"options": [
			{
				"label": "feat",
				"description": "A new feature"
			},
			{
				"label": "wip",
				"description": "Work in progress"
			},
			{
				"label": "fix",
				"description": "A bug fix"
			},
			{
				"label": "config",
				"description": "Configuration changes"
			},
			{
				"label": "docs",
				"description": "Documentation changes"
			},
			{
				"label": "chore",
				"description": "Updating dependencies, remove TODOs, etc."
			}
		]
	},
	{
		"label": "Scope",
		"name": "scope",
		"description": "A scope may be provided to a commit’s type, to provide additional contextual information and is contained within parenthesis, e.g., <code>feat(parser): add ability to parse arrays</code>.",
		"type": "enum",
		"prefix": "(",
		"suffix": ")",
		"options": [
			{
				"label": "content",
				"description": "App content (code)"
			},
			{
				"label": "env",
				"description": "App environment (NextJs, Vercel, Typescript, documentation, etc.)"
			},
			{
				"label": "infra",
				"description": "Infrastructure (vscode, extensions, etc.)"
			}
		]
	},
	{
		"label": "Gitmoji",
		"name": "gitmoji",
		"description": "Visual indicator for the change",
		"type": "enum",
		"options": [
			{
				"label": ":sparkles:",
				"description": "✨ Introduce new feature"
			},
			{
				"label": ":art:",
				"description": "🎨 Improve / refactor structure of the code"
			},
			{
				"label": ":construction:",
				"description": "🚧 Work in progress"
			},
			{
				"label": ":bug:",
				"description": "🐛 Fix a bug"
			},
			{
				"label": ":lipstick:",
				"description": "💄 Add or update styling"
			},
			{
				"label": ":wrench:",
				"description": "🔧 Add or update configuration"
			},
			{
				"label": ":memo:",
				"description": "📝 Add or update documentation"
			},
			{
				"label": ":arrow_up:",
				"description": "⬆️ Adjust dependencies"
			},
			{
				"label": ":wheelchair:",
				"description": "♿️ Improve accessibility"
			},
			{
				"label": ":zap:",
				"description": "⚡️ Improve performance"
			},
			{
				"label": ":lock:",
				"description": "🔒 Improve security"
			},
			{
				"label": ":mag:",
				"description": "🔍 Improve SEO"
			}
		]
	},
	{
		"label": "Short description",
		"name": "description",
		"description": "Short description in the subject line.",
		"type": "text",
		"multiline": false
	},
	{
		"label": "Body",
		"name": "body",
		"description": "Optional body",
		"type": "text",
		"multiline": true
	},
	{
		"label": "Breaking change",
		"name": "breaking_change",
		"type": "boolean",
		"value": "BREAKING CHANGE: ",
		"default": false
	},
	{
		"label": "Footer",
		"name": "footer",
		"description": "Optional footer",
		"type": "text",
		"multiline": true
	}
],
```

**Comment by bendera** [comment]
Hi,

I've updated the select component and that may cause the error. I will fix it as I can.

**Comment by bendera** [comment]
Until I fix it, you can install another version:
https://stackoverflow.com/a/66134500/2314863

**Comment by TheThirdRace** [comment]
Awesome, thanks for the very prompt answer!

**Comment by TheThirdRace** [comment]
@bendera 
As a follow up, I did what you suggested and installed a previous version.

But now I'm a bit confused... I could swear the `<select>` would be filled with the first value and I wouldn't need to change it for it to apply automatically, but that's not what I'm seeing now with version 0.18.1...

It's possible I was mistaken in my original post and I did have to select the values manually after all.

With version 0.18.1, the `<select>` has no default values so I need to select one. Since I change the value from no selection to a selected value, I only have 1 step to fill the `<select>`.

With version 0.18.2, since the first value is pre-filled, I have to change the value twice to be able to select the first value.

In any case, there's still a problem with twice the amount of step required to select the first value.

The best outcome would be that the pre-selected value would just work, so no need to change the selected value and it would work.

The second best outcome would be back to the previous version behavior, where the `<select>` wouldn't have a pre-selected value and thus, simply changing the value once would work.

I hope this clarifies the issue a bit. If you have further question, don't hesitate to ask.
